core-libs/setup/ssr/engine/cx-common-engine.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  CommonEngine,
+  CommonEngineOptions,
+  CommonEngineRenderOptions,
+} from '@angular/ssr';
+import { PROPAGATE_ERROR_TO_SERVER } from '../error-handling/error-response/propagate-error-to-server';
+
+/**
+ * The Spartacus extension of the CommonEngine introduced to handle propagated server responses caught during server-side rendering.
+ * @extends {CommonEngine}
```

Comment:
```suggestion
 * The Spartacus extension of the Angular's `CommonEngine`. It is able to handle the propagated server responses caught during server-side rendering of a Spartacus app.
 * For reference, see Angular's source code: https://github.com/angular/angular-cli/blob/6cf866225ab09f8b4b3803c000b632bed8448ce4/packages/angular/ssr/src/common-engine.ts#L56
 *
 * @extends {CommonEngine}
```
https://github.com/angular/angular-cli/blob/6cf866225ab09f8b4b3803c000b632bed8448ce4/packages/angular/ssr/src/common-engine.ts#L56